File: exercism/clojure/clock/src/clock.clj
(ns clock)

(defn clock [a b]
  (let [q (Math/floorDiv b 60)
        h (mod (+ a q) 24)
        m (mod b 60)]
    {:hours h :minutes m}))

(defn add-time [c n]
  (clock (:hours c) (+ (:minutes c) n)))

(defn clock->string [c]
  (format "%02d:%02d" (:hours c) (:minutes c)))

File: exercism/haskell/clock/src/Clock.hs
module Clock (clockHour, clockMin, fromHourMin, toString) where

import Text.Printf (printf)

newtype Clock = C { _minutes :: Int } deriving (Eq)

_minsPerDay, _minsPerHr, _hrsPerDay :: Int
_minsPerDay = _minsPerHr * _hrsPerDay
_minsPerHr  = 60
_hrsPerDay  = 24

instance Num Clock where
  fromInteger n = C $ fromIntegral n
  negate (C m)  = C $ (-m) `mod` _minsPerDay
  (C x) + (C y) = C $ x + y `mod` _minsPerDay
  (*)           = undefined
  abs           = undefined
  signum        = undefined

instance Show Clock where
  show = toString

clockHour :: Clock -> Int
clockHour = (`mod`_hrsPerDay) . (`div`_minsPerHr) . _minutes

clockMin :: Clock -> Int
clockMin = (`mod`_minsPerHr) . _minutes

fromHourMin :: Int -> Int -> Clock
fromHourMin hour min = C $ (_minsPerHr * hour + min) `mod` _minsPerDay

toString :: Clock -> String
toString = printf "%02d:%02d" <$> clockHour <*> clockMin
